File: src/main/java/hu/elte/alkfejl/hirportal/entity/Article.java
package hu.elte.alkfejl.hirportal.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.sql.Timestamp;

import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "ARTICLE")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Article {

    @JoinColumn
    @ManyToOne(targetEntity = User.class, optional = false)
    @JsonIgnoreProperties("articles")
    private User user;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    @Column(name="title", nullable=false, length=100)
    private String title;

    @Column(name="author", nullable=true, length=100)
    private String author;

    @Column(name="date", nullable = false)
    private Timestamp date;

    @Column(name="body", columnDefinition="CLOB NOT NULL")
    private String body;

    @Column(name="thumbnail", nullable=false, length=200)
    private String thumbnail;

    @Column(name="category", nullable=false, length=100)
    private int category;

}

File: src/app/article.ts
import { User } from './user';

export class Article {
    public id = 0;
    public author: String = '';
    public date: number = Date.now();
    public title: String = '';
    public thumbnail: String = '';
    public body: String = '';
    public category = 1;
    public user: User = new User;

    /*constructor(id?: number, author?: String, date?: number, title?: String, thumbnail?: String, body?: String) {
        this.id = id || 0;
        this.author = author || '';
        this.date = date || Date.now();
        this.title = title || '';
        this.thumbnail = thumbnail || '';
        this.body = body || '';
    }*/
}
